src/site/content/en/blog/url-protocol-handler/index.md

```diff
@@ -0,0 +1,267 @@
+   uninstallation).
+
+## Demo
+
+You can see a demo of URL protocol handler registration for PWAs on Glitch.
+
+1. Go to [https://protocol-handler.glitch.me/](https://protocol-handler.glitch.me/), install the
+   PWA, and reload the app after the installation. The browser has now registered the PWA as a
+   handler for the `web+coffee` protocol with the operating system.
+1. In the installed PWA window, click on the link
+   [https://protocol-handler-link.glitch.me/](https://protocol-handler-link.glitch.me/). This will
+   open a new browser tab with three links. Click on the first or the second (latte macchiato or
+   americano). The browser will now show you a prompt and ask if you are fine with the app being a
+   protocol handler for the `web+coffee` protocol. If you agree, the PWA will open and show the
+   selected coffee.
+1. To compare with the traditional flow that uses `navigator.registerProtocolHandler()`, click
+   the \*_Register protocol handler_ button in the PWA. Then in the browser tab click the third link
+   (chai). It will likewise show a prompt, but then open the PWA in a tab, not in a browser window.
+1. Send yourself a message on a platform-specific application like Skype on Windows
+   with a link like `<a href="web+coffee://americano">Americano</a>` and click it. It should
+   likewise open the installed PWA.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/lfjgVAiGjVE9VohX3EDs.png", alt="URL protocol handler demo with browser tab with links on the left, and standalone PWA window on the right.", width="800", height="461" %}
+
+## Security considerations
+
+Since PWA installation requires the context to be secure, protocol handling inherits this
+constraint. The list of registered protocol handlers is not exposed to the web in any way so it
+cannot be used as a fingerprinting vector.
+
+### Non user-initiated navigation attempts
+
+Navigation attempts that are not initiated by the user, but that are programmatic, may not open
+apps. The custom protocol URL may only be used in top-level browsing contexts, but not, for example,
+as the URL of an iframe.
+
+### Allowlist of protocols
+
+Just like with `registerProtocolHandler()` there is an allowlist of protocols that apps can register
+to handle.
+
+### Default protocol handlers
+
+Registration of PWA protocol handlers will not take over the default handler for a protocol.
+Instead, the next time the protocol is invoked, an operating system disambiguation dialog will
+prompt the user to either keep using the default handler or select the newly registered handler.
+
+### Consent prompt
+
+On the first launch of the PWA due to an invoked protocol, the user will be presented with a
+permission dialog. This dialog will display the app name and origin of the app, and ask the user if
+the app is allowed to handle links from the protocol. If a user rejects the permission dialog, the
+protocol handler is unregistered with the operating system.
```

Comment:
This is outdated. The only way to unregister protocols for web apps is by uninstalling the app. Cancelling the consent prompt merely closes it. The user consent will be saved in prefs if they click allow.

![image](https://user-images.githubusercontent.com/1686833/117515587-8e50cc80-af4b-11eb-905a-78ef7f9a87fb.png)

Reply:
I have updated this accordingly.